src/resources/parser/glb-parser.js

```diff
@@ -1195,6 +1223,130 @@ var createNode = function (gltfNode, nodeIndex) {
+};
+
+var createModel = function (name, meshGroup, materials, defaultMaterial) {
+    var model = new Model();
+    model.graph = new GraphNode(name);
+
+    meshGroup.forEach(function (mesh) {
+        var material = (mesh.materialIndex === undefined) ? defaultMaterial : materials[mesh.materialIndex];
+        var meshInstance = new MeshInstance(model.graph, mesh, material);
+
+        if (mesh.morph) {
+            var morphInstance = new MorphInstance(mesh.morph);
+            if (mesh.weights) {
+                for (var wi = 0; wi < mesh.weights.length; wi++) {
+                    morphInstance.setWeight(wi, mesh.weights[wi]);
+                }
+            }
+
+            meshInstance.morphInstance = morphInstance;
+            model.morphInstances.push(morphInstance);
+        }
+
+        model.meshInstances.push(meshInstance);
+    });
+
+    return model;
+};
+
+var createCamera = function (gltfCamera, node) {
+    var cameraProps = {
+        enabled: false
+    };
+
+    if (gltfCamera.type === "orthographic") {
+        const orthographic = gltfCamera.orthographic;
+        const xMag = orthographic.xmag;
+        const yMag = orthographic.ymag;
+        const aspectRatio = xMag !== undefined ? xMag / yMag  : undefined;
+
+        Object.assign(cameraProps, {
+            projection: PROJECTION_ORTHOGRAPHIC,
+            aspectRatioMode: aspectRatio !== undefined ? ASPECT_MANUAL : ASPECT_AUTO,
+            aspectRatio: aspectRatio,
+            orthoHeight: yMag,
+            farClip: orthographic.zfar,
+            nearClip: orthographic.znear
+        });
+    } else {
+        const perspective = gltfCamera.perspective;
+        const aspectRatio = perspective.aspectRatio;
+
+        Object.assign(cameraProps, {
+            projection: PROJECTION_PERSPECTIVE,
+            aspectRatioMode: aspectRatio !== undefined ? ASPECT_MANUAL : ASPECT_AUTO,
+            aspectRatio: aspectRatio,
+            fov: perspective.yfov * math.RAD_TO_DEG,
+            farClip: perspective.zfar,
+            nearClip: perspective.znear
+        });
+    }
+
+    return node.addComponent("camera", cameraProps);
+};
+
+var createLight = function (gltfLight, node) {
+    var lightProps = {
+        type: gltfLight.type,
+        falloffMode: LIGHTFALLOFF_INVERSESQUARED
+    };
+
+    if (gltfLight.hasOwnProperty('color')) {
+        lightProps.color = new Color(...gltfLight.color);
+    }
+
+    if (gltfLight.hasOwnProperty('intensity')) {
+        // TODO: Tweak intensity to match glTF specification. Point and spot lights use luminous
+        // intensity in candela (lm/sr) while directional lights use illuminance in lux (lm/m2).
+        lightProps.intensity = gltfLight.intensity;
+    } else {
+        lightProps.intensity = 1;
+    }
+
+    if (gltfLight.hasOwnProperty('range')) {
+        lightProps.range = gltfLight.range;
+    } else {
+        // TODO: Better way of representing infinite range
+        lightProps.range = 99999999;
```

Comment:
What happens if we actually use `Number.POSITIVE_INFINITY`?

Reply:
I did not know of `Number.POSITIVE_INFINITY`, will give it a try